### PR TITLE
Fixed eval for checkboxes

### PR DESCRIPTION
**Pull request by beckyconning**
Part of a fix for https://slamdata.atlassian.net/browse/SD-1302

### CODE REVIEW

**Comment by jonsterling** [comment]
Could you summarize what the old behavior actually was (because I'm curious) and what the new behavior is? thanks! :smile:

**Comment by garyb** [comment]
I can explain since I helped with this:

- The two lists a `CheckBoxes` constructor has should zip together so that the list of booleans corresponds to the options.
- Previously the list of selections was constructed by mapping over each of the values returned by the eval result for the selections, and then checking if they each was present in the list of options. This means the selections list had no real connection to the options list.
- We now map over the options list and check if each value is present in the evaluated selections list instead, ensuring the length and order of the boolean list corresponds to the options list.

**Comment by garyb** [comment]
@beckyconning I reassigned this to @jonsterling since I'm "too close" to review it really :smile:

**Comment by jonsterling** [comment]
@garyb thank you for summarizing!

@beckyconning Do we need to update the "spec" in the README?

**Comment by beckyconning** [comment]
Yes. That could be a separate PR as it was something that was already wrong but I think it makes just as much sense to do it here.

**Comment by jdegoes** [comment]
Indeed, and we need an update to the docs in slamdata/slamdata.github.io which cover SlamDown.

**Comment by jonsterling** [comment]
@beckyconning OK, I think I'm :+1: for this! Thanks for digging into this issue.

@jdegoes Is this the document that needs updating? https://github.com/slamdata/slamdata.github.io/blob/master/_posts/documentation/2015-11-02-slamdown-reference.md

**Comment by jdegoes** [comment]
I pulled up three markdown files:

<https://github.com/slamdata/slamdata.github.io/search?utf8=✓&q=slamdown>

The index.html file doesn't contain any meaningful information, it's just a link.

**Comment by jonsterling** [comment]
OK, cool. thanks @jdegoes
